ksp_plugin_adapter/plotter.cs

```diff
@@ -0,0 +1,243 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+
+namespace principia {
+namespace ksp_plugin_adapter {
+
+class Plotter {
+  public Plotter(PrincipiaPluginAdapter adapter) {
+    adapter_ = adapter;
+  }
+
+  public void Plot(DisposablePlanetarium planetarium,
+                   string main_vessel_guid,
+                   double history_length) {
+    PlotCelestialTrajectories(planetarium, main_vessel_guid, history_length);
+    if (main_vessel_guid == null) {
+      return;
+    }
+    PlotVesselTrajectories(planetarium, main_vessel_guid, history_length);
+  }
+
+  private void PlotVesselTrajectories(DisposablePlanetarium planetarium,
+                                      string main_vessel_guid,
+                                      double history_length) {
+    planetarium.PlanetariumPlotPsychohistory(
+        Plugin,
+        main_vessel_guid,
+        history_length,
+        VertexBuffer.data,
+        VertexBuffer.size,
+        out vertex_count_);
+    DrawLineMesh(psychohistory_, adapter_.history_colour, adapter_.history_style);
+    planetarium.PlanetariumPlotPrediction(
+        Plugin,
+        main_vessel_guid,
+        VertexBuffer.data,
+        VertexBuffer.size,
+        out vertex_count_);
+    DrawLineMesh(prediction_, adapter_.prediction_colour, adapter_.prediction_style);
+
+    // Target psychohistory and prediction.
+    string target_id = FlightGlobals.fetch.VesselTarget?.GetVessel()?.id.
+        ToString();
+    if (FlightGlobals.ActiveVessel != null &&
+        !adapter_.plotting_frame_selector_.target_frame_selected &&
+        target_id != null &&
+        Plugin.HasVessel(target_id)) {
+      planetarium.PlanetariumPlotPsychohistory(
+          Plugin,
+          target_id,
+          history_length,
+          VertexBuffer.data,
+          VertexBuffer.size,
+          out vertex_count_);
+      DrawLineMesh(target_psychohistory_, adapter_.target_history_colour,
+                   adapter_.target_history_style);
+      planetarium.PlanetariumPlotPrediction(
+          Plugin,
+          target_id,
+          VertexBuffer.data,
+          VertexBuffer.size,
+          out vertex_count_);
+      DrawLineMesh(target_prediction_, adapter_.target_prediction_colour,
+                   adapter_.target_prediction_style);
+    }
+    // Main vessel flight plan.
+    if (Plugin.FlightPlanExists(main_vessel_guid)) {
+      int number_of_segments =
+          Plugin.FlightPlanNumberOfSegments(main_vessel_guid);
+      for (int i = flight_plan_segments_.Count; i < number_of_segments; ++i) {
```

Comment:
`flight_plan_segments_` is a poor name, it makes you think that it is the segments themselves.  Call it `flight_plan_segment_meshes_` and explain in the declaration that this is a buffer that grows dynamically to the maximum number of segments that we have to paint.